demo/kserve/custom-manifests/caikit/tgis-servingruntime-template.yaml

```
@@ -0,0 +1,58 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+labels:
+  app: opendatahub
+  template: tgis-serving-grpc
+metadata:
+  annotations:
+    description: Text Generation Inference Server (TGIS) is a high performance inference engine that deploys and serves Large Language Models.
+    openshift.io/display-name: TGIS Standalone ServingRuntime for KServe
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: rhods,rhoai,kserve,servingruntime
+    template.openshift.io/documentation-url: https://github.com/opendatahub-io/text-generation-inference
+    template.openshift.io/long-description: This template defines resources needed to deploy TGIS standalone servingruntime with KServe in Red Hat OpenShift AI
+    template.openshift.io/support-url: https://access.redhat.com
+  labels:
+    app: opendatahub
+  name: tgis-serving-grpc-template
+objects:
+- apiVersion: serving.kserve.io/v1alpha1
+  kind: ServingRuntime
+  metadata:
+    name: tgis-runtime-grpc
+  spec:
+    multiModel: false
+    supportedModelFormats:
+      - autoSelect: true
+        name: pytorch
+    containers:
+      - name: kserve-container
+        image: ${TGIS_CONTAINER_IMAGE}
+        command: ["text-generation-launcher"]
+        args: 
+          - "--model-name=/mnt/models/"
+          - "--port=3000"
+          - "--grpc-port=8033"
+        env:
+          - name: TRANSFORMERS_CACHE
+            value: /tmp/transformers_cache
+        resources: 
+           requests:
+             cpu: ${CPU_REQUEST}
+             memory: ${MEMORY_REQUEST}
+        ports:
```

Probes for tgis as done in #186 

```suggestion
      readinessProbe: # Use exec probes instad of httpGet since the probes' port gets rewritten to the containerPort
        exec:
          command:
            - curl
            - localhost:3000/health
        initialDelaySeconds: 5
      livenessProbe:
        exec:
          command:
            - curl
            - localhost:3000/health
        initialDelaySeconds: 5
        # periodSeconds: 5
        ports:
```